docs: add Enterprise Data API reference and usage reports guide (#25383)
docker/docker.github.io: content/manuals/enterprise/reports/_index.md

---
title: Usage reports
description: Learn how to retrieve enterprise usage reports for your Docker organization using the Reports API.
keywords: docker, enterprise, reports, usage, pulls, api, csv, organization access token, oat
weight: 20
params:
  sidebar:
    group: Enterprise
---

Docker provides daily usage reports for organizations with a Docker Business
subscription. These reports contain pull activity data for your organization and
are available as CSV downloads through the Reports API.

Reports are generated automatically. You use the API to list what is available
and download the files you need.

## Prerequisites

Before you begin, ensure you have:

- A [Docker Business subscription](/subscription/details/)
- One of the following:
  - Organization owner role
  - A custom role that includes the **report-read** permission

> - `curl` installed for making API requests

Reviewer: ```suggestion
```

> - `jq` installed for JSON parsing (optional, for formatting responses)

Reviewer: ```suggestion

```

> ## Authentication

Reviewer: ```suggestion
## Set up OAT for Reports API
```

> The Reports API requires an Organization Access Token (OAT). OATs are

Reviewer: Move into pre-reqs to replace curl/jq prereqs

Reviewer: Specifically some guidance around having available OATs, since some orgs have limitations; they need an open OAT slot to create a new one (if I'm understanding correctly)

org-scoped tokens designed for machine-to-machine access, making them
suitable for automated report retrieval workflows. Personal Access Tokens

> (PATs) are not supported.

Reviewer: turn into tip call out in prereqs

1. [Create an organization access token](/enterprise/security/access-tokens/)
   and select the **Report Read** scope under Organization permissions.

2. Set your variables:

   ```bash
   ORG="<your-org-name>"
   TOKEN="<your-organization-access-token>"
   ```

3. Test the token:

   ```console
   $ curl -s "https://api.docker.com/enterprise-data/v1/orgs/$ORG/reports" \
     -H "Authorization: Bearer $TOKEN" | jq .
   ```

You use this `TOKEN` value in the `Authorization: Bearer` header for all
subsequent API calls.

> ## List available report types

Reviewer: ```suggestion
## List reports
```

> Discover which report types and cadences are available for your organization.

Reviewer: ```suggestion
You can fetch report types and their cadences on a per-organization basis.  For example, you can fetch all available reports to your organization, or you can fetch a scoped set of reports by time interval. 
```

```console
$ curl -s "https://api.docker.com/enterprise-data/v1/orgs/$ORG/reports" \
  -H "Authorization: Bearer $TOKEN" | jq .
```

Example response:

```json
{
  "report_types": [
    {
      "ReportType": "usage_pulls",
      "Cadence": "daily"
    }
  ]
}
```

Each entry represents a distinct combination of report type and cadence. Use
these values in subsequent calls.

> ## List reports

Reviewer: ```suggestion
### List reports by type and cadence
```

> List available reports for a given type and cadence. Reports are returned in

Reviewer: ```suggestion
Based off the response to the list available reports query, you can list reports by type and cadence.  Reports are returned in
```

> reverse chronological order (most recent first).

Reviewer: ```suggestion
reverse chronological order so the most recent are listed first.
```

$ curl -s "https://api.docker.com/enterprise-data/v1/orgs/$ORG/reports/usage_pulls/daily" \
  -H "Authorization: Bearer $TOKEN" | jq .
```

Example response:

```json
{
  "reports": [
    {
      "ReportType": "usage_pulls",
      "Cadence": "daily",
      "Date": "2026-06-16",
      "SizeBytes": 48210,
      "Key": "my-org/usage_pulls/daily/2026-06-16.csv"
    },
    {
      "ReportType": "usage_pulls",
      "Cadence": "daily",
      "Date": "2026-06-15",
      "SizeBytes": 51003,
      "Key": "my-org/usage_pulls/daily/2026-06-15.csv"
    }
  ],
  "next_page_token": ""
}
```

### Pagination

Results are paginated with a default page size of 30 and a maximum of 100.
Use the `page_size` and `page_token` query parameters to control pagination.

```console
$ curl -s "https://api.docker.com/enterprise-data/v1/orgs/$ORG/reports/usage_pulls/daily?page_size=10" \
  -H "Authorization: Bearer $TOKEN" | jq .
```

When `next_page_token` is non-empty, pass it as `page_token` to fetch the next
page:

```console
$ curl -s "https://api.docker.com/enterprise-data/v1/orgs/$ORG/reports/usage_pulls/daily?page_size=10&page_token=NEXT_TOKEN" \
  -H "Authorization: Bearer $TOKEN" | jq .
```

> ## Download a report

Reviewer: ```suggestion
## Download reports
```

Reviewer: From a ToC POV, we want these specific actions to be nested and balanced for better visibility. We should add a sentence at minimum that describes why it's best practice to download a report, what you might use it for, and generalized information that's scattered in the below h3s. for example, this would be the place we say reports are downloaded as CSVs.
